lib/ToolbarPanelHub.jsm

```diff
@@ -219,81 +243,142 @@ class _ToolbarPanelHub {
-        relatedToCurrent: true,
-        triggeringPrincipal: Services.scriptSecurityManager.createNullPrincipal(
-          {}
-        ),
-        csp: null,
-      });
-    });
-    const titleEl = this._createElement(doc, "h2");
-    titleEl.classList.add("whatsNew-message-title");
-    this._setString(doc, titleEl, content.title);
-    wrapperEl.appendChild(titleEl);
 
-    const bodyEl = this._createElement(doc, "p");
-    this._setString(doc, bodyEl, content.body);
-    wrapperEl.appendChild(bodyEl);
+    this._attachClickListener(win, wrapperEl, message);
 
-    if (content.link_text) {
-      const linkEl = this._createElement(doc, "a");
-      linkEl.classList.add("text-link");
-      this._setString(doc, linkEl, content.link_text);
-      wrapperEl.appendChild(linkEl);
+    wrapperEl.appendChild(
+      this._createElement(doc, "h2", {
+        classList: "whatsNew-message-title",
+        content: message.content.title,
+      })
+    );
+    wrapperEl.appendChild(
+      this._createElement(doc, "p", { content: message.content.body })
+    );
+
+    if (message.content.link_text) {
+      wrapperEl.appendChild(
+        this._createElement(doc, "a", {
+          classList: "text-link",
+          content: message.content.link_text,
+        })
+      );
     }
 
     return messageEl;
   }
 
-  _createElement(doc, elem) {
-    return doc.createElementNS("http://www.w3.org/1999/xhtml", elem);
+  _createElement(doc, elem, options = {}) {
+    const node = doc.createElementNS("http://www.w3.org/1999/xhtml", elem);
+    if (options.classList) {
+      node.classList.add(options.classList);
+    }
+    if (options.content) {
+      this._setString(doc, node, options.content);
+    }
+
+    return node;
   }
 
-  _createDateElement(doc, date) {
-    const dateEl = this._createElement(doc, "p");
-    dateEl.classList.add("whatsNew-message-date");
-    dateEl.textContent = new Date(date).toLocaleDateString("default", {
-      month: "long",
-      day: "numeric",
-      year: "numeric",
-    });
-    return dateEl;
+  async _contentArguments() {
+    let nf = new Services.intl.NumberFormat();
+    // Between now and 6 weeks ago
+    const dateTo = new Date();
+    const dateFrom = new Date(dateTo.getTime() - 42 * 24 * 60 * 60 * 1000);
+    const eventsByDate = await TrackingDBService.getEventsByDateRange(
+      dateFrom,
+      dateTo
+    );
+    // Count all events in the past 6 weeks
+    const totalEvents = eventsByDate.reduce(
+      (acc, day) => acc + day.getResultByName("count"),
+      0
+    );
+    return {
+      // Keys need to match variable names used in asrouter.ftl
+      // `earliestDate` will be either 6 weeks ago or when tracking recording
+      // started. Whichever is more recent.
+      earliestDate: new Date(
+        Math.max(
+          new Date(await TrackingDBService.getEarliestRecordedDate()),
+          dateFrom
+        )
+      ).getTime(),
+      blockedCount: nf.format(totalEvents),
```

Comment:
Is this the new / expected way to do things? Otherwise plain JS `totalEvents.toLocaleString()` should work ?

Reply:
Good point this is straightforward.
I knew Fluent formats numbers and I just looked into using their API.